### ISSUE TITLE
Missing `quay.io/skopeo/stable:v1.11.0` image

### ISSUE THREAD

**Issue by mattclay** [containers/skopeo#1895]
I see that [v11.0.0](https://github.com/containers/skopeo/releases/tag/v1.11.0) has been released, but the image is [missing](https://quay.io/repository/skopeo/stable?tab=tags) from quay.io.

The latest image available is v1.10.0, which was updated on quay.io about 2 hours ago.

**Comment by mtrmac** [comment]
Thanks for your report. Per https://github.com/containers/skopeo/tree/main/contrib/skopeoimage , the stable repo contains Fedora packages that are considered stable by Fedora.

That Fedora process is ongoing: https://bodhi.fedoraproject.org/updates/?packages=skopeo 

Cc: @lsm5

**Comment by lsm5** [comment]
@mattclay it's gonna take a little longer, I just found out another issue that was causing gating test failures. Once I have the new builds, I'll update here. please add karma.

**Comment by lsm5** [comment]
Please add karma to https://bodhi.fedoraproject.org/updates/FEDORA-2023-11be4d0146 so it can move to stable faster (and get pushed to quay).

**Comment by mattclay** [comment]
I see the `stable:v1.11` image exists now, but there's no `stable:v1.11.0` -- is that to be expected?

**Comment by mtrmac** [comment]
Oops.

So what seems to have happened here is that the `v1.11.0` tag is placed slightly incorrectly, and points at a version that identifies itself, in `skopeo --version`, as `v11.1.1-dev`; and that’s the image tag that now exists on `quay.io`.

---

One question: what to do about that incorrect tag.
- We probably could move the `v1.11.0` tag in the git repo; I think that’s mostly a bad idea.
- We could change the Fedora packaging to use the right commit, and report the right version, but it would not match the upstream tag. That also seems surprising and risky.
- Practically speaking, I think this is going to be much less of a concern with the upcoming `v1.11.1` release. Assuming we make that one correctly.

So I think leaving `v1.11.0` be and waiting for the next release seems easiest.

---

Another question: is there a reasonable way to prevent such a mistake?
- Some kind of Git (GitHub?) hook to reject tags with mismatching versions?
- A `make` target to create the version bump commits and tags correctly?

Arguably, _one_ instance of a process mistake after so many tags (79 now) might not be a good reason to worry about this.

But then, we have a very similar versioning process across those repos, so automating this a bit more for all those projects might pay off.

Cc: @rhatdan @cevich

**Comment by cevich** [comment]
There's no check/control possible once the tag is pushed - unlesss the tag-pushing itself is done via a script, then (obviously) having some checks is possible.

A `make` target would be sexy, but how would it know what kind of release you're doing.  I think you'd need at least three targets, `major`, `minor`, and `patch`.  Then you'd need some way to translate those targets into updates in a file - the real tricky part.

IMHO, I think this can be called a "woopsie" and we move on.  I can delete the image on quay, and as long as the code is fixed, it should not re-appear.  Is this okay?

**Comment by mtrmac** [comment]
(Something like `make release new-tag='1.20'` might be possible, or just maintain a shell script.)

> IMHO, I think this can be called a "woopsie" and we move on. 

I’m fine with that.

> I can delete the image on quay,

I’d somewhat prefer waiting until we actually have a `1.11.1`, so that the structure of the repo (individually-tagged images + latest-x.y tag) is preserved.

**Comment by cevich** [comment]
Okay.  How should we remember that, is having this issue open enough?

**Comment by mtrmac** [comment]
I think that, + a Gmail sleep feature in my inbox, could work.

**Comment by mtrmac** [comment]
https://github.com/containers/skopeo/releases/tag/v1.11.1 was just published.

**Comment by lsm5** [comment]
Created  https://bodhi.fedoraproject.org/updates/FEDORA-2023-33cfd04d6c . Will take about a week to land in the stable repos. Please add karma if possible.

**Comment by cevich** [comment]
Following up: I searched both skopeo registries on quay but found nothing tagged `v11*`.  With the actual `v1.11.1` release, the `latest` and tagged images will fix themselves as soon as packages go out to `updates` and `updates-testing`.

**Comment by mtrmac** [comment]
@cevich I see an `v1.11.1-dev` on https://quay.io/repository/skopeo/stable?tab=tags . (Sure, that’s not ideal because the tag doesn’t have the expected form; still, you said “nothing tagged”.)

**Comment by cevich** [comment]
Oh! Good catch I missed that one.  I'll nuke it.

**Comment by mtrmac** [comment]
OTOH removing a tag might break existing deployments… and won’t the automation recreate it (in one of the repos, at least) within 24 hours anyway, until 1.11.1 is actually in Fedora stable?

**Comment by cevich** [comment]
It's too late, I already removed it.  Yes once it hits 'updates' a `v1.11.1` tagged image will show up.  I think it's not worth worrying about (unless the image doesn't appear).

**Comment by mtrmac** [comment]
Circling back, there is now a `v1`, `v1.11`, `v1.11.1` and `latest`, all pointing at the same image.

(There is, also, a `v1.11.1-dev` tag, created after the previous manual removal.)
